src/test/java/com/softserve/academy/spaced/repetition/controller/CourseControllerTest.java

```diff
@@ -0,0 +1,286 @@
+package com.softserve.academy.spaced.repetition.controller;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.softserve.academy.spaced.repetition.domain.*;
+import com.softserve.academy.spaced.repetition.service.CourseService;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CourseControllerTest {
+    private MockMvc mockMvc;
+
+    @InjectMocks
```

Comment:
Remind me, please, what is this line for?

Reply:
Another useful approach is to not start the server at all, but test only the layer below that, where Spring handles the incoming HTTP request and hands it off to your controller. That way, almost the full stack is used, and your code will be called exactly the same way as if it was processing a real HTTP request, but without the cost of starting the server. To do that we will use Spring’s MockMvc, and we can ask for that to be injected for us.

Reply:
Yes, thank you, I understand, but it makes me confused. As far as I understood @InjectMocks injects mocks instead of all the fields of CourseController but why do we need a @Mock annotation near the CourseService then if it's already injected by @InjectMocks annotation?

Reply:
@ Mock near the CourseService is required to be able to manipulate the mock, for example, to write:
`when(courseService.getCourseById(eq(categoryId),eq(courseId))).thenReturn(createCourse());`

Reply:
Thank you, now it's clear.